fastly/base_fastly_service.go

```diff
@@ -544,11 +549,16 @@ func resourceServiceDelete(_ context.Context, d *schema.ResourceData, meta inter
 		}
 	}
 
-	err := conn.DeleteService(&gofastly.DeleteServiceInput{
-		ID: d.Id(),
-	})
-	if err != nil {
-		return diag.FromErr(err)
+	if !d.Get("reuse").(bool) {
```

Comment:
Would it be clearer to make this a check for force_destroy instead of not reuse?

Reply:
No, because if the user doesn't specify anything then this breaks the current behaviour.

e.g. currently, if the user has a deactivated service and they don't specify `force_destroy`, then the service will successfully be deleted. By changing the condition to be `d.Get("force_destroy").(bool)` we're effectively saying they _have_ to add `force_destroy` to delete their deactivated service. By negating the reuse check we can keep the existing behaviour while supporting the new behaviour.